src/base-switch/base-switch.tsx

```diff
@@ -61,13 +66,30 @@ export const BaseSwitch = React.forwardRef<HTMLInputElement, BaseSwitchProps>(
       [setCheckedState],
     );
 
-    const onInputFocus = useCallback(() => {
-      setIsInputFocused(true);
-    }, [setIsInputFocused]);
+    const onInputFocus = useCallback(
+      (e: React.FocusEvent<HTMLInputElement>) => {
+        if (isFocusVisible(e)) {
+          setIsInputFocusedVisible(true);
+        }
```

Comment:
is it the same if we do 
```suggestion
   setIsInputFocusedVisible(isFocusVisible(e)); 
```

Reply:
Yup, nice one.